sample.tex

```diff
@@ -54,12 +54,18 @@
 
 % graphics packages
 \usepackage[pdftex]{graphicx} % remove pdftex if you are not compiling to pdf
-\graphicspath{{./figures/}} % this places all graphics in the figures subdirectory
+\graphicspath{{./figures/}{../figures/}} % this places all graphics in the figures subdirectory
 
 % allowed graphics extensions
 % uncomment if you prefer to add extension in \includegraphics
 \DeclareGraphicsExtensions{.pdf,.png,.jpg}
 
+% allows the document to be broken into subfiles and compiled individually
+% note that if you place the subfiles in a separate directory, you need to place
+% a copy of thesis.cls in that directory. There should be a way to fix this, but
+% I was forced to move on before figuring it out.
+\usepackage{subfiles}
+
```

Comment:
I'm a little hesitant to include this.  for a few reasons:

1.  I think we would need to solve the problem of including the thesis.cls in the subdirectory, I would prefer not to have duplicate code
2.  Users can always look up this feature.  Although it is nice to include useful things, there does have to be some limit because if we begin including too many things it may become confusing / cluttered.  That said, it is always difficult to know where that line should be drawn.  I'm open to hearing arguments as to why it should be included, but am leaning toward leaving this one out.
3.  if we do use subfiles in the example, I think it would be cleaner to simply follow that paradigm instead of allowing both ways, which would mean a bit of refactoring here and that all users would then use subfiles by default.  Might this be a bit much for the sample doc?

Reply:
another thought... in my dissertation I use something like this to include subfiles:
```
\input{intro.tex}
\input{background.tex}
\input{methods.tex}
```
although it doesn't allow things to be compiled individually, it does break things up while not depending on an additional package

Reply:
I enjoy `subfile` because I frequently re-compile, and so the time savings adds up quickly for me. Given the messiness introduced by duplicating the `thesis.cls` file and the extra refactoring it would take to switch the template over to such a model, I think it makes sense to allow users to add this at their own discretion.